refactor(posix): replace `echo -n`, etc with `printf` (`shellcheck` SC3037)

## Skills required

- command line
- shell scripting (i.e. bash, posix)
- git (obviously)

## What to do

1. Clone and enter the repo
   ```sh
   git clone https://github.com/webinstall/webi-installers.git ./webi-installers/
   cd ./webi-installers/
   ```
2. Install [`fd`](https://webinstall.dev/fd) and [`shellcheck`](https://webinstall.dev/shellcheck):
   ```sh
   curl -sS https://webinstall.dev/fd | bash
   curl -sS https://webinstall.dev/shellcheck | bash
   ```
3. Follow the on-screen instructions (or close and re-open the terminal)
   ```txt
   CLOSE AND RE-OPEN YOUR TERMINAL!
   ```
4. Find and fix the offending uses of `echo -n` (or echo with other flags). \
    (you may find [`sd`](https://webinstall.dev/sd) useful to automate some of the find-and-replace work)
   ```sh
   fd -t f -e sh | \
       xargs shellcheck -s sh -S warning -i SC3037 --
   ```

--- a/_webi/template.sh
+++ b/_webi/template.sh
@@ -118,9 +118,9 @@ __bootstrap_webi() {
             # but that's okay, 'cmp -s' is good enough for us
             if cmp -s "${pkg_src_cmd}" "${my_current_cmd}"; then
                 echo "${my_canonical_name} already installed:"
-                echo -n "    ${pkg_dst}"
+                printf "    %s" "${pkg_dst}"
                 if [ "${pkg_src_cmd}" != "${my_current_cmd}" ]; then
-                    echo -n " => ${pkg_src}"
+                    printf " => %s" "${pkg_src}"
                 fi
                 echo ""
                 exit 0
@@ -345,7 +345,7 @@ __bootstrap_webi() {
     __init_installer() {
 
         # do nothing - to satisfy parser prior to templating
-        echo -n ""
+        printf ""
 
         # {{ installer }}
 

--- a/git-config-gpg/git-config-gpg.sh
+++ b/git-config-gpg/git-config-gpg.sh
@@ -17,11 +17,11 @@ __git_gpg_init() {
         gpg-pubkey-id
     )"
 
-    echo -n "Enabling automatic git commit signing...
-	    git config --global user.signingkey ${MY_KEY_ID}
+    printf "Enabling automatic git commit signing...
+	    git config --global user.signingkey %s
 	    git config --global commit.gpgsign true
 	    git config --global log.showSignature true
-	"
+	" "${MY_KEY_ID}"
 
     git config --global user.signingkey "${MY_KEY_ID}"
     git config --global commit.gpgsign true

--- a/gpg-pubkey/gpg-pubkey.sh
+++ b/gpg-pubkey/gpg-pubkey.sh
@@ -70,7 +70,7 @@ _create_gpg_key() {
         echo >&2 "Choose a passphrase for this GPG Key."
         echo >&2 "(the passphrase will not be shown as you type)"
         echo >&2 ""
-        echo >&2 -n "Passphrase: "
+        printf >&2 "Passphrase: "
         MY_REPLY="$(__shadow_read)"
         echo >&2 ""
         echo "

